File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int m=x;
        int flag=0;
        if(m<0){
            flag=1;
        }
        long sum=0;
        while(x>0||x<0){
            int rem=x%10;
            sum=sum*10+rem;
            if(sum<Integer.MIN_VALUE||sum>Integer.MAX_VALUE){
                return 0;
            }
            x=x/10;
        }
        return (int) sum;
    }
}

File: 0007-reverse-integer/0007-reverse-integer.c
int reverse(int x){
    int rem=0;
    long long sum=0;
while(x>0||x<0){
    rem=x%10;
    sum= (long long)sum*10 +rem;
     if (sum > INT_MAX || sum < INT_MIN) {
            return 0;
        }
    x=x/10;
}
return sum;
}